caffe2/operators/elementwise_ops.h

```diff
@@ -147,19 +147,19 @@ class BinaryElementwiseWithArgsOp final : public Operator<Context> {
   bool DoRunWithType() {
     const auto& A = Input(0);
     const auto& B = Input(1);
-    auto* C = Output(0);
+
     const T* A_data = A.template data<T>();
     const T* B_data = B.template data<T>();
     std::vector<int> A_dims;
     std::vector<int> B_dims;
+    std::vector<int64_t> C_dims;
 
     if (legacy_broadcast_) {
-      CAFFE_ENFORCE_NE(
-          C,
-          &B,
+      CAFFE_ENFORCE(
+          !IsInputOutputAlias(1, 0),
           "In-place is allowed only with the first tensor when "
```

Comment:
this wasn't changed in this diff, but this message is weird. if it is legacy broadcast, then it asserts that it is not inplace, saying "inplace is only allowed if legacy broadcast"? Is this a bug?

Reply:
looks like this is asserting we can only allow Output(0) being alias of Input(0), this is checking Output(0) is not an alias of Input(1). cc @BIT-silence